tests/test_cmab_meta_model.py

```
@@ -228,6 +249,121 @@ def test_backbone_update_changes_backbone_and_heads(
+        ids=["l2_reg_factor_zero", "backbone_lr_none"],
+    )
+    def test_backbone_training_knob_disabled_matches_default(
+        self, explicit_kwarg: dict, rng: np.random.Generator
+    ) -> None:
+        """Explicitly disabling a backbone training knob reproduces the default (knob-unset) behavior exactly."""
+        actions, rewards, context = self._balanced_batch(rng)
+        default_meta = self._build_meta(with_backbone=True)
+        explicit_meta = self._build_meta(with_backbone=True, **explicit_kwarg)
+        default_meta.update(actions=actions, rewards=rewards, context=context)
+        explicit_meta.update(actions=actions, rewards=rewards, context=context)
+        for w_default, w_explicit in zip(default_meta.backbone.weight_arrays, explicit_meta.backbone.weight_arrays):
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)